keps/sig-apps/0034-maxunavailable-for-statefulsets.md

```diff
@@ -0,0 +1,179 @@
+published by these new pod starts a new time series which makes tracking metrics for the application difficult. While this could be mitigated, 
+it requires some tricks on the time series collection side. It would be so much better, If we could use a StatefulSet object so my object names doesnt 
+change and hence all metrics goes to a single time series. This will be easier if StatefulSet is at feature parity with Deployments.
+2: My Container does some initial startup tasks like loading up cache or something that takes a lot of time. If we used StatefulSet, we can only go one 
+pod at a time which would result in a slow rolling update. If we did maxUnavailable for StatefulSet with a greater than 1 number, it would allow for a 
+faster rollout.
+3: My Stateful clustered application, has followers and leaders, with followers being many more than 1. My application can tolerate many followers going 
+down at the same time. I want to be able to do faster rollouts by bringing down 2 or more followers at the same time. This is only possible if StatefulSet
+supports maxUnaavailable in Rolling Updates.
+4: Sometimes i just want easier tracking of revisions of a rolling update. Deployment does it through ReplicaSets and has its own nuances. Understanding 
+that requires diving into the complicacy of hashing and how replicasets are named. Over and above that, there are some issues with hash collisions which 
+further complicate the situation(I know they were solved). StatefulSet introduced ControllerRevisions in 1.7 which I believe are easier to think and reason 
+about. They are used by DaemonSet and StatefulSet for tracking revisions. It would be so much nicer if all the use cases of Deployments can be met and we 
+could track the revisions by ControllerRevisions.
+
+With this feature in place, when using StatefulSet with maxUnavailable >1, the user understands that this would not cause issues with their Stateful 
+Applications which have per pod state and identity while still providing all of the above written advantages.
+
+### Goals
+StatefulSet RollingUpdate strategy will contain an additional parameter called `maxUnavailable` to control how many Pods will be brought down at a time,
+during Rolling Update.
+
+### Non-Goals
+maxUnavailable is only implemeted to affect the Rolling Update of StatefulSet. Considering maxUnavailable for Pod Management Policy is beyond the purview
+of this KEP.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a User of Kubernetes,  I can create a StatefulSet with RollingUpdate Strategy and specify maxUnavailable.
+
+### Implementation Details
+
+#### API Changes
+
+Following changes will be made to the Rolling Update Strategy for StatefulSet.
+
+```go
+// RollingUpdateStatefulSetStrategy is used to communicate parameter for RollingUpdateStatefulSetStrategyType.
+type RollingUpdateStatefulSetStrategy struct {
+        // Partition indicates the ordinal at which the StatefulSet should be
+        // partitioned.
```

Comment:
An example here will be clearer.
You can add example of the 5 total replicas you wrote below here.

Reply:
sorry i should have made this clearer. this is not a new field but an existing field.